File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/common/WaitTask.java
package org.firstinspires.ftc.teamcode.common;

public class WaitTask extends BaseTask implements Task {


    public WaitTask(BunyipsOpMode opMode, double time) {
        super(opMode, time);

    }

    void update() {

    }

    @Override
    public void run() {
        if (isFinished()) {
            return;
        }

    }

}

File: TeamCode/BunyipsLib/src/main/java/org/murraybridgebunyips/bunyipslib/tasks/WaitTask.kt
package org.murraybridgebunyips.bunyipslib.tasks

import org.murraybridgebunyips.bunyipslib.BunyipsOpMode

/**
 * Task to wait for a specific amount of time.
 */
class WaitTask(private val opMode: BunyipsOpMode, time: Double) : Task(time), RobotTask {
    override fun init() {
        return
    }

    override fun run() {
        opMode.addTelemetry("Waiting % seconds...", time)
        opMode.idle()
    }

    override fun isTaskFinished(): Boolean {
        return false
    }

    override fun onFinish() {
        return
    }
}